nexus/src/app/sagas/region_snapshot_replacement_step.rs

```diff
@@ -423,6 +424,21 @@ async fn rsrss_notify_upstairs(
     let params = sagactx.saga_params::<Params>()?;
     let log = sagactx.user_data().log();
 
+    // If the associated volume was deleted, then skip this notification step.
+    // It's likely there is no Upstairs to talk to, but continue with the saga
```

Comment:
"Likely"? :)

I wonder if it'd be worthwhile to have a `#[cfg(debug_assertions)]` block here in which we assert that if the volume is deleted, then either it has no disk or the disk is detached.

Reply:
a78cabf45137705fa8645a7799ed4afcf310d7cd reworks the comment.

I tried your suggestion by adding a ``validate_higher_level_resource_deleted` function to the `validate_volume_invariants` check, but it returned an Err during the `disk_create` saga unwind. I'm not sure how to solve this, or even if it can be solved...?